Element.js

```diff
@@ -257,9 +257,20 @@ Element.prototype = {
 	 * @returns {Promise.<void>}
 	 */
 	type: function (value) {
+		function arrayToText(array) {
+			var textValue = array;
+			while (Array.isArray(textValue)) {
+	            textValue = array.join('');
+	        }
+	        return textValue;
+	    }
+
 		function getPostData(arrayValue) {
-			if (self.session.capabilities.isWebDriver) {
+			if (self.session.capabilities.isWebDriver &&
+				!self.session.capabilities.valueParameterCalledText) {
```

Comment:
Was there a case where `isWebDriver` was true and `valueParameterCalledText` was also true? In theory, `isWebDriver` should cover that case, because if something is using WebDriver semantics it should be using `value`.

Reply:
Yes. In the case where the post command fails and then getPostData is called again after setting the flag. If I leave off that conditional, then the array will be split improperly causing the value to fail because a string is expected, not an array of characters. Geckodriver 16.0+